drivers/accel/adxl355/adxl355.c

```diff
@@ -169,8 +174,21 @@ int adxl355_init(struct adxl355_dev **device,
 
 	ret = adxl355_read_device_data(dev, ADXL355_ADDR(ADXL355_PARTID),
 				       GET_ADXL355_TRANSF_LEN(ADXL355_PARTID),&reg_value);
-	if (ret || (reg_value != GET_ADXL355_RESET_VAL(ADXL355_PARTID)))
+	if (ret)
+		goto error_com;
+
+	switch(dev->dev_type) {
```

Comment:
you could get rid of this switch with a simple array where index is ID_ADXL... which contains PARTID's.